src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/loading.ts

```diff
@@ -0,0 +1,34 @@
+// Can be used in various components to check whether an API call is waiting for a response for loading spinners
+// Example in Application.vue we do this in a button :loading="loadingStore.isLoading('application_post')"
+// Loading states are set by our apiResultHandler.ts where we provide a key based on the API operations we have ex. application_post
+
+import { defineStore } from "pinia";
+
+import type { OperationMethods } from "@/types/openapi";
+
+export type LoadingOperation = keyof OperationMethods;
+
+export type LoadingState = {
+  [key in LoadingOperation]: boolean;
+};
+
+export interface LoadingStateConfig {
+  loadingStates: LoadingState;
+}
+
+export const useLoadingStore = defineStore("loading", {
+  state: (): LoadingStateConfig => ({
+    loadingStates: {} as LoadingState,
+  }),
+
+  getters: {
+    isLoading() {
+      return (key: LoadingOperation) => this.loadingStates[key];
```

Comment:
Do we need to check if loadingStates has the key? I think you can set the isLoading function to return type which may then complain that `this.loadingStates[key]` can return `undefined`

i.e.
```
  getters: {
    isLoading(): boolean {
      return (key: LoadingOperation) => this.loadingStates[key];
      ...
```

Reply:
yeah it maintains all the other keys. I tested this by adding another property when we hit submit "draftapplication_put"

and I see that the key stays there 

![image](https://github.com/bcgov/ECC-ECER/assets/74216496/460d3cc6-4a3d-48b9-845e-3b352af9c5ca)